### ISSUE TITLE
DeploymentManager::classloaders are not being closed when doing vertx.close()

### ISSUE THREAD

**Issue by tomfi** [eclipse-vertx/vert.x#3002]
While running tests in my projects, I start a **Vertx** instance **BeforeEach** and deploy the relevant verticles that i want to test with isolation group and extra-classpath (we use that to create a plugin-like system with isolated classpaths) and close the vertx instance **AfterEach**.

Some of the tests fetch a **plugin** (shaded vertx jar with a main-verticle), place it in a folder, and load it dynamically with an isolation group and extra-classpath. After calling **vertx.close()** I tried to remove the temp jar file but it was used by the jvm and could not be deleted.

After debugging it, I found that the reason it is being held by the JVM is that inside **VertxImpl** there is **DeploymentManager** that has **Map<String, ClassLoader> classloaders** and it does not close the URLClassLoader instances when calling **vertx.close()**.

To test my assumption, i did some ugly reflection work to check it:
```kotlin
suspend fun Vertx.closeAllAwait() {
  this.closeAwait()

  val deploymentManagerField = this::class.java.getDeclaredField("deploymentManager")
  deploymentManagerField.isAccessible = true
  val deploymentManager = deploymentManagerField.get(this) as DeploymentManager

  val classLoadersField = deploymentManager.javaClass.getDeclaredField("classloaders")
  classLoadersField.isAccessible = true
  val classLoaders = classLoadersField.get(deploymentManager) as Map<String, ClassLoader>

  classLoaders
    .forEach {
      try {
        val classLoader: URLClassLoader = it.value as URLClassLoader
        classLoader.close()
      } catch (e: Exception) {
        // ignore!
      }
    }
}
```

After using my new **vertx.closeAllAwait()** the jar could be removed as the classloaders created by vertx were closed.

I wonder if it is not closing the classloaders by design and if it is possible to close them when vertx is closing, or maybe add a boolean flag and close them when the user want to.

**EDIT:** it is a bit inconsistent, sometimes the classloaders list is empty but the JVM still hold the file, but i still wonder if some other resources in vertx are not being released.

**Comment by vietj** [comment]
I think this should be improved and created classloaders should use close when the classloader implements Closeable

**Comment by vietj** [comment]
this should be fixed in 3.8 and master, would you mind to check ? https://github.com/eclipse-vertx/vert.x/issues/3017

**Comment by tomfi** [comment]
@vietj sure, ill check it first thing in the morning. Thanks for the quick reply and fix :)
